tests/src/kakarot/instructions/test_block_information.cairo

```diff
@@ -39,203 +39,33 @@ func constructor{
     return ();
 }
 
-@view
-func test__blockhash_should_push_blockhash_to_stack{
-    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}() {
-    // Given # 1
-    alloc_locals;
-    let fp_and_pc = get_fp_and_pc();
-    local __fp__: felt* = fp_and_pc.fp_val;
-
-    let (bytecode) = alloc();
-    let stack = Stack.init();
-
-    let (local block_number_: Uint256) = block_number.read();
-
-    let stack = Stack.push(stack, &block_number_);
-    let ctx = TestHelpers.init_context_with_stack(0, bytecode, stack);
-
-    // When
-    let result = BlockInformation.exec_blockhash(ctx);
-
-    // Then
-    assert result.gas_used = 20;
-    assert result.stack.size = 1;
-    let (stack, index0) = Stack.peek(result.stack, 0);
-    let (blockhash_) = blockhash.read();
-    let value = Helpers.to_uint256(val=blockhash_);
-    assert_uint256_eq([index0], [value]);
-
-    // Given # 2
-    let (bytecode) = alloc();
-    let stack = Stack.init();
-
-    let (current_block_number: felt) = get_block_number();
-    let out_of_range_block = current_block_number - 260;
-    assert_nn(out_of_range_block);
-    let out_of_range_block_uint256 = Helpers.to_uint256(val=out_of_range_block);
-    let stack = Stack.push(stack, out_of_range_block_uint256);
-    let ctx = TestHelpers.init_context_with_stack(0, bytecode, stack);
-
-    // When
-    let result = BlockInformation.exec_blockhash(ctx);
-
-    // Then
-    assert result.gas_used = 20;
-    assert result.stack.size = 1;
-    let (stack, index0) = Stack.peek(result.stack, 0);
-    assert index0.low = 0;
-    assert index0.high = 0;
-
-    return ();
-}
-
-@view
-func test__chainId__should_push_chain_id_to_stack{
-    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}() {
-    // Given
-    alloc_locals;
-    let (bytecode) = alloc();
-    let ctx = TestHelpers.init_context(0, bytecode);
-
-    // When
-    let result = BlockInformation.exec_chainid(ctx);
-
-    // Then
-    assert result.gas_used = 2;
-    assert result.stack.size = 1;
-    let (stack, index0) = Stack.peek(result.stack, 0);
-    assert index0.low = 1263227476;
-    assert index0.high = 0;
-    return ();
-}
-
 @external
-func test__coinbase_should_push_coinbase_address_to_stack{
+func test__exec_block_information{
     syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}() {
+}(opcode: felt, stack_len: felt, stack: Uint256*) -> (
+    result: Uint256, timestamp: felt, block_number: felt
+) {
     // Given
     alloc_locals;
     let (bytecode) = alloc();
-    let ctx = TestHelpers.init_context(0, bytecode);
+    assert [bytecode] = opcode;
+    let stack_ = Stack.init();
+    if (stack_len != 0) {
+        let stack_ = Stack.push_uint256(stack_, stack[0]);
```

Comment:
as I understand, here stack_len is at most 1?
should we slightly refactor to make it clearer?